relay/circuit-v2.md

```diff
@@ -0,0 +1,398 @@
+# Circuit Relay v2
+
+This is the version 2 of the libp2p Circuit Relay protocol.
+
+| Lifecycle Stage | Maturity       | Status | Latest Revision |
+|-----------------|----------------|--------|-----------------|
+| 1A              | DRAFT          | Active | r0, 2021-05-05  |
+
+Authors: [@vyzo]
+
+Interest Group: [@mxinden], [@stebalien], [@raulk]
+
+[@vyzo]: https://github.com/vyzo
+[@mxinden]: https://github.com/mxinden
+[@stebalien]: https://github.com/stebalien
+[@raulk]: https://github.com/raulk
+
+See the [lifecycle document][lifecycle-spec] for context about maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+## Table of Contents
+
+- [Introduction](#introduction)
+  - [Rationale](#rationale)
+- [The Protocol](#the-protocol)
+  - [Interaction](#interaction)
+  - [Hop Protocol](#hop-protocol)
+  - [Stop Protocol](#stop-protocol)
+  - [Reservation Vouchers](#reservation-vouchers)
+- [Protobuf](#protobuf)
+
+## Introduction
+
+This is the specification of v2 of the p2p-circuit relay protocol.
+
+Compared to the first version of the protocol, there are some significant departures:
+- The protocol has been split into two subprotocols, `hop` and `stop`
+  - The `hop` protocol governs the behaviour of relays; it is used for
```

Comment:
I don't understand "governs the behaviour of relays". I think you intend to say that it's the protocol with which clients send commands to the relay (RESERVE, CONNECT).

```suggestion
  - The `hop` protocol is client-initiated, and is used when clients send commands
    to relays; it is used for
```